sympy/core/mul.py

```diff
@@ -1255,19 +1255,20 @@ def _eval_is_integer(self):
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
         if is_rational:
-            if d is S.One:
+            if d == S.One:
                 return True
             elif d == S(2):
                 return n.is_even
-        # if d is even -- 0 or not -- the
```

Comment:
Please leave this here since it is the rationale for not handling d.is_zero separately. In general, if you didn't change the code then don't change the comments unless you make them more clear or they are clearly wrong.

Reply:
I forgot to put em back. sorry

Reply:
np :-)